### PR TITLE
Ask for Bitbucket project key instead of options box

### LINKED ISSUE #116: More Than 100 Bitbucket Projects
Slack will not show an options box with more than 100 options. Falling back to entering the project key for now. Later #109 will be done to restore functionality when there are fewer than 100 projects

### PR DESCRIPTION
**Pull request by chriskieser**
Note ticket #109 restores this options list for Bitbucket project selection when there are fewer than 100 projects

fixes #116

### CODE REVIEW

File: src/gluon/bitbucket/BitbucketProject.ts [hunk 1]
```diff
@@ -81,70 +81,49 @@ export class ListExistingBitbucketProject implements HandleCommand<HandlerResult
...
+        return gluonMemberFromScreenName(ctx, this.screenName)
+            .then(member => {
+                return gluonProjectFromProjectName(ctx, this.projectName)
+                    .then(gluonProject => {
+                        // get the selected project's details
+                        const projectRestUrl = `${QMConfig.subatomic.bitbucket.restUrl}/api/1.0/projects/${this.bitbucketProjectKey}`;
+                        const projectUiUrl = `${QMConfig.subatomic.bitbucket.baseUrl}/projects/${this.bitbucketProjectKey}`;
+                        return bitbucketAxios().get(projectRestUrl)
+                            .then(project => {
+                                return axios.put(`${QMConfig.subatomic.gluon.baseUrl}/projects/${gluonProject.projectId}`,
```
Comment (kbristow) [inline src/gluon/bitbucket/BitbucketProject.ts]: This works? Would have thought you would need to use the rest url...

File: src/gluon/bitbucket/BitbucketProject.ts [hunk 2]
```diff
@@ -81,70 +81,49 @@ export class ListExistingBitbucketProject implements HandleCommand<HandlerResult
...
+                                }, this.teamChannel);
+                            })
+                            .catch(error => {
+                                if (error.response && error.response.status === 404) {
+                                    return ctx.messageClient.addressChannels({
+                                        text: `⚠️ The Bitbucket project with key ${this.bitbucketProjectKey} was not found`,
+                                    }, this.teamChannel)
+                                        .then(failure);
+                                } else {
+                                    throw error;
```
Comment (kbristow) [inline src/gluon/bitbucket/BitbucketProject.ts]: Shouldn't throw errors in the promise chain. Rather fail the promises. Something like `return failure(error);`

File: src/gluon/bitbucket/BitbucketProject.ts (hunk 1 above)
Comment (chriskieser) [reply src/gluon/bitbucket/BitbucketProject.ts]: Seems to work. Was from the existing code, not something new

File: src/gluon/bitbucket/BitbucketProject.ts (hunk 2 above)
Comment (chriskieser) [reply src/gluon/bitbucket/BitbucketProject.ts]: Good point. Fixed

**Pull request merged**